src/rinoh/__main__.py

```diff
@@ -216,13 +216,21 @@ def main():
         parser.print_help()
         return
 
+    if not os.path.exists(args.input):
+        raise SystemExit('{}: No such file'.format(args.input))
+    input_dir, input_filename = os.path.split(args.input)
+    input_root, input_ext = os.path.splitext(input_filename)
+
     template_cfg = {}
     variables = {}
-    cwd_source = CwdSource()
+    input_source = InputDirSource(input_dir)
     if args.stylesheet:
-        if os.path.isfile(args.stylesheet):
-            stylesheet = StyleSheetFile(args.stylesheet, matcher=matcher,
-                                        source=cwd_source)
+        stylesheet_path = Path(args.stylesheet)
+        if not stylesheet_path.is_absolute():
+            stylesheet_path = input_source.root / stylesheet_path
+        if stylesheet_path.exists():
+            stylesheet = StyleSheetFile(str(stylesheet_path.resolve()), matcher=matcher,
+                                        source=input_source)
```

Comment:
I haven't yet tested this, but does this mean that specifying a style sheet file as a path relative to the CWD is no longer supported? I think we should restore this former behavior (which I broke).

The new behavior this PR introduces is to search for style sheet and template configuration files relative to the input file. I'm not sure that is intuitive, since these files are not necessarily specific to a single input document. Personally, I would expect `rinoh` to, at least, first search the CWD.

For reference, consider the builder for Sphinx, which interprets values  for _template_ as a path relative to the location of `conf.py`. For a command line tool like `rinoh`, I think that the CWD should be the reference for relative paths.

Reply:
I failed to understand the requirements before writing the code... rookie error. Rather than reverting everthing and having a messy history I will close this PR and create a new one with the relevant changes.

Reply:
Sorry, this was indeed not clear from the bug ticket. It's a always good idea to discuss specifics before starting work on a PR.